libraries/apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/HttpCacheExtensions.kt

```diff
@@ -74,18 +74,19 @@ fun ApolloClient.Builder.httpCache(
       maxSize = maxSize,
   )
   val apolloRequestToCacheKey = mutableMapOf<String, String>()
-  var apolloRequestUuid: Uuid? = null
   return addHttpInterceptor(object : HttpInterceptor {
     override suspend fun intercept(request: HttpRequest, chain: HttpInterceptorChain): HttpResponse {
       val cacheKey = CachingHttpInterceptor.cacheKey(request)
-      apolloRequestToCacheKey[apolloRequestUuid!!.toString()] = cacheKey
+      val requestUuid = request.headers.valueOf(CachingHttpInterceptor.REQUEST_UUID_HEADER)!!
+      synchronized(apolloRequestToCacheKey) {
+        apolloRequestToCacheKey[requestUuid] = cacheKey
+      }
       return chain.proceed(request.newBuilder().addHeader(CachingHttpInterceptor.CACHE_KEY_HEADER, cacheKey).build())
```

Comment:
Filter out `REQUEST_UUID_HEADER` here so that it's not seen by the backend?

Reply:
Good catch!